File: src/main/java/unaverage/strategic_ench/config/BlacklistEnchantment.java
package unaverage.strategic_ench.config;

import com.google.common.collect.ImmutableSet;
import com.typesafe.config.Config;
import net.minecraft.enchantment.Enchantment;
import net.minecraft.util.registry.Registry;
import unaverage.strategic_ench.Helper;

import java.util.HashSet;
import java.util.Set;

/**
 * Contains configuration value relating to whether enchantments are blacklisted from appearing within the game
 */
public class BlacklistEnchantment {
    static Set<String> getDefault(boolean newFile){
        if (!newFile) return Set.of();

        return ImmutableSet.of(
            "minecraft:protection"
        );
    }

    final Set<String> current;

    BlacklistEnchantment(Config c) {
        current = new HashSet<>(
            c.getStringList(this.getClass().getSimpleName())
        );
    }

    /**
     * @param e The enchantment being tested
     * {@return True if the enchantment is blacklisted from appearing from the game, false otherwise}
     */
    public boolean isBlackListed(Enchantment e){
        return Helper.contains(
            e,
            current,
            Registry.ENCHANTMENT
        );
    }
}

File: src/main/kotlin/unaverage/strategic_ench/config/BlacklistEnchantment.kt
package unaverage.strategic_ench.config

import com.google.common.collect.ImmutableSet
import net.minecraft.enchantment.Enchantment

/**
 * Contains configuration value relating to whether enchantments are blacklisted from appearing within the game
 */
class BlacklistEnchantment internal constructor() {
    val current: Set<String>

    init {
        current = HashSet()
    }

    /**
     * @param e The enchantment being tested
     * {@return True if the enchantment is blacklisted from appearing from the game, false otherwise}
     */
    fun isBlackListed(e: Enchantment?): Boolean {
        return false
    }

    companion object {
        fun getDefault(newFile: Boolean): Set<String> {
            return if (!newFile) java.util.Set.of() else ImmutableSet.of(
                "minecraft:protection"
            )
        }
    }
}